app/src/main/java/io/github/djunicode/canteenapp/BaseActivity.java

```diff
@@ -1,21 +1,72 @@
 package io.github.djunicode.canteenapp;
 
+import android.content.Intent;
 import android.content.SharedPreferences;
 import android.os.Bundle;
 import android.os.PersistableBundle;
 import android.support.annotation.Nullable;
 import android.support.v7.app.AppCompatActivity;
+import android.widget.Toast;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import io.github.djunicode.canteenapp.ResponseObjects.SignInResponse;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
 
 import static io.github.djunicode.canteenapp.LoginActivity.TOKEN_SHARED_PREFS_KEY;
 import static io.github.djunicode.canteenapp.LoginActivity.TOKEN_STRING;
 
 public abstract class BaseActivity extends AppCompatActivity {
 
     String tokenReturn;
+    protected Retrofit customRetrofit;
+    protected Boolean addToken = true;
+    private String toast;
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        toast = this.getIntent().getStringExtra("toastMessage");
+        if (toast != null) {
+            Toast.makeText(getApplicationContext(), toast, Toast.LENGTH_SHORT).show();
+        }
+
+        OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
+        httpClient.connectTimeout(60, TimeUnit.SECONDS);
+        httpClient.readTimeout(60, TimeUnit.SECONDS);
+        httpClient.writeTimeout(60, TimeUnit.SECONDS);
+        httpClient.addInterceptor(new Interceptor() {
+            @Override
+            public Response intercept(Chain chain) throws IOException {
+
+                Request request = chain.request();
+                Response response = chain.proceed(request);
+                if(response.code() == 401){
+                    removeToken();
+                    Intent intent = new Intent(getApplicationContext(), LoginActivity.class).putExtra(
+                            "toastMessage", "Security Credentials Invalid"
+                    );
+                    startActivity(intent);
+                    return response;
+                }
+                return response;
+            }
+        });
+        customRetrofit = new Retrofit.Builder()
```

Comment:
Please create a separate Java file for defining the retrofit instance as you will be needing it in multiple places

Reply:
We can extend base activity in all other activities so the same instance can be used without putting in a separate class